BenchPress/Helpers/BenchPress.Azure/Common.psm1

```diff
@@ -121,19 +126,202 @@ function Get-ResourceByType {
   }
-  else{
+  else {
     return Get-AzResource -Name "${ResourceName}" -ResourceGroupName "${ResourceGroupName}"
   }
 }
 
-Export-ModuleMember -Function Get-Resource, Get-ResourceByType
+
+<#
+  .SYNOPSIS
+    Confirms whether a resource exists or properties on a resource are configured correctly.
+
+  .DESCRIPTION
+    The Confirm-AzBPResource cmdlet confirms whether an Azure resource exists and/or confirms whether properties
+    on a resource exist and are configured to the correct value. The cmdlet will return a ConfirmResult object
+    which contains the following properties:
+    - Success: True if the resource exists and/or the property is set to the expected value. Otherwise, false.
+    - Error: System.Management.Automation.ErrorRecord containing an ErrorRecord with a message explaining
+             that a resource did not exist or property was not set to the expected value.
+    - ResourceDetails: System.Object that contains the details of the Azure Resource that is being confirmed.
+
+  .PARAMETER ResourceName
+    The name of the Resource
+
+  .PARAMETER ResourceGroupName
+    The name of the Resource Group
+
+  .PARAMETER ResourceType
+    The type of the Resource (currently support the following:
+    ActionGroup
+    AKSCluster
+    AppServicePlan
+    ContainerRegistry
+    KeyVault
+    ResourceGroup
+    SqlDatabase
+    SqlServer
+    VirtualMachine
+    WebApp)
+
+  .PARAMETER PropertyKey
+    The name of the property to check on the resource
+
+  .PARAMETER PropertyValue
+    The expected value of the property to check
+
+  .EXAMPLE
+    Checking whether a resource exists (i.e. Resource Group)
+    Confirm-AzBPResource -ResourceType $resourceType -ResourceName $resourceGroupName
+
+  .EXAMPLE
+    Confirm whether a resource has a property configured correctly (i.e. Resource Group located in West US 3)
+    Confirm-AzBPResource -ResourceType $resourceType -ResourceName $resourceGroupName -PropertyKey "Location" `
+                         -PropertyValue "WestUS3"
+  .INPUTS
+    ResourceType
+    System.String
+
+  .OUTPUTS
+    ConfirmResult
+#>
+function Confirm-Resource {
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [ResourceType]$ResourceType,
+
+    [Parameter(Mandatory = $true)]
+    [string]$ResourceName,
+
+    [Parameter(Mandatory = $false)]
+    [string]$ResourceGroupName,
+
+    [Parameter(Mandatory = $false)]
+    [string]$PropertyKey,
+
+    [Parameter(Mandatory = $false)]
+    [string]$PropertyValue
+
+  )
+
+  $ConfirmResult = [ConfirmResult]::new()
+  $ConfirmResult.ResourceDetails = Get-ResourceByType -ResourceGroupName $ResourceGroupName -ResourceName $ResourceName -ResourceType $ResourceType
+
+  if ($null -ne $ConfirmResult.ResourceDetails) {
+    $ConfirmResult.Success = $true
+  }
+  else {
+    $ConfirmResult.Success = $false
+    $ConfirmResult.Error = Format-NotExistError -Expected $ResourceName
+    return $ConfirmResult
```

Comment:
nit: I'm a bit averse to multiple returns from a method. This could be addressed in a couple of ways:

1. Remove this return and move the `if ($PropertyKey)` block inside the `if ($null -ne $ConfirmResult.ResourceDetails)` block.
2. Remove this return and change the below `if` statement to `$ConfirmResult.Success -and $PropertyKey`.